src/test/java/com/restapi/carMarket/service/CarServiceUnitTest.java

```diff
@@ -11,168 +8,199 @@
 
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.Mockito;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
+import org.mockito.junit.MockitoJUnitRunner;
 import org.springframework.data.domain.Example;
-import org.springframework.test.context.junit4.SpringRunner;
+
 
 import java.util.Optional;
 
+
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.BDDMockito.given;
 
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(classes = CarMarketApplication.class)
+@RunWith(MockitoJUnitRunner.class)
 public class CarServiceUnitTest {
-    @Autowired
-    CarService carService;
-
-    @MockBean
+    @Mock
     CarDao carDAO;
 
+    @InjectMocks
+    CarService carService;
+
     @Test
-    public void insertionOfValidCarShouldReturnTrue() throws Exception{
+    public void insertionOfValidCarShouldReturnTrue() {
         Car car = new Car("BMW","5 Series", 2017, 40000);
-        carService.insert(car);
+        carService.addCarToMarket(car);
         Mockito.verify(carDAO,Mockito.times(1)).save(car);
```

Comment:
`times()` is optional in this case because it's same as default. Saying `Mockito.verify(carDao).save(car)` verifies that the method is called exactly once.

Reply:
Done. Removed unneeded code from all methods that were using this call.